common/shell-local/run.go

```diff
@@ -27,6 +28,24 @@ type EnvVarsTemplate struct {
 }
 
 func Run(ui packer.Ui, config *Config) (bool, error) {
+	// Check if shell-local can even execute against this runtime OS
+	runCommand := false
+	if len(config.OnlyOn) > 0 {
+		for _, os := range config.OnlyOn {
+			if os == runtime.GOOS {
+				runCommand = true
+				break
+			}
+		}
+	} else {
+		runCommand = true
+	}
+	if !runCommand {
+		ui.Say(fmt.Sprintf("Skipping shell-local due to runtime OS"))
+		log.Printf("[INFO] (shell-local): skipping shell-local due to missing runtime OS")
+		return true, nil
+	}
+
```

Comment:
I think we can still simplify this just a bit, since we needn't parse anything if OnlyOn isn't set: 

```
	if len(config.OnlyOn) > 0 {
        	runCommand := false
		for _, os := range config.OnlyOn {
			if os == runtime.GOOS {
				runCommand = true
				break
			}
		}
		if !runCommand {
			ui.Say(fmt.Sprintf("Skipping shell-local due to runtime OS"))
			log.Printf("[INFO] (shell-local): skipping shell-local due to missing runtime OS")
			return true, nil
		}		
	} 
```

Reply:
I like it! Clean and condensed.